cpp/src/decisiontree/decisiontree.cu

```diff
@@ -159,8 +159,20 @@ void decisionTreeClassifierFit(const raft::handle_t &handle,
                                uint64_t seed) {
   std::shared_ptr<DecisionTreeClassifier<float>> dt_classifier =
     std::make_shared<DecisionTreeClassifier<float>>();
+  MLCommon::device_buffer<float> *global_quantiles = nullptr;
```

Comment:
minor nitpick: please use `std::unique_ptr` instead to automatically take care of memory lifetimes. Then you wouldn't also need to do the explicit `release` and `delete` below.